build-js.sh

```diff
@@ -3,12 +3,113 @@
 # to run this, as the builds are bundled in the repo in bin/. Running this is
 # useful if you are a developer and want to update those builds.
 #
+# Usage: build-js.sh
+# Usage: EMSCRIPTEN=path/to/emscripten build-js.sh  # explicit emscripten dir
+#
+# Emscripten's em++ and tools/webidl_binder.py will be accessed through the
+# env var EMSCRIPTEN, e.g. ${EMSCRIPTEN}/em++
+#
+# You can get emscripten from
+# http://kripken.github.io/emscripten-site/docs/getting_started/downloads.html
+#
+set -e
+
+if [ "$1" == "-h" ] || [ "$1" == "--help" ] || [ "$1" == "-help" ]; then
+  echo "usage: $0 [-g]" >&2
+  echo "-g  produce debug build" >&2
+  exit 1
+fi
+
+if [ "$EMSCRIPTEN" == "" ]; then
+  echo "$0: EMSCRIPTEN environment variable is not set" >&2
+  exit 1
+fi
+
+if [ ! -d "$EMSCRIPTEN" ]; then
+  echo "$0: $EMSCRIPTEN is not a directory" >&2
```

Comment:
This is wrong. `$EMSCRIPTEN` will be expanded in the message to the empty string. I'll change this to `\$EMSCRIPTEN`